app/src/main/java/com/example/socialmediavbsanalay/presentation/fragments/MainPageFragment.kt

```diff
@@ -0,0 +1,196 @@
+package com.example.socialmediavbsanalay.presentation.fragments
+
+import android.app.AlertDialog
+import android.content.ActivityNotFoundException
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import android.provider.MediaStore
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.core.content.ContextCompat.checkSelfPermission
+import androidx.fragment.app.Fragment
+import androidx.navigation.Navigation
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.socialmediavbsanalay.R
+import com.example.socialmediavbsanalay.databinding.FragmentMainPageBinding
+import com.example.socialmediavbsanalay.presentation.adapters.PostAdapter
+import com.example.socialmediavbsanalay.presentation.adapters.StoryAdapter
+import dagger.hilt.android.AndroidEntryPoint
+import android.Manifest
+import android.content.Context
+import androidx.activity.result.contract.ActivityResultContracts
+
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [MainPageFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+@AndroidEntryPoint
+class MainPageFragment : Fragment() {
+
+    private var _binding: FragmentMainPageBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var storyAdapter: StoryAdapter
+    private lateinit var postAdapter: PostAdapter
+    private val REQUEST_IMAGE_CAPTURE = 1
```

Comment:
memory efficient olması açısından class'ın en altında bir companion object oluşturup bu değerleri companion object içerisinde örneğin:

private const val REQUEST_IMAGE_CAPTURE = 1 olarak tanımlar mısın

Reply:
çözüldü